internal/app/agent/register/register.go

```diff
@@ -0,0 +1,50 @@
+package register
+
+import (
+	"strings"
+
+	nats "github.com/nats-io/go-nats"
+)
+
+const (
+	HEARTBEAT = "heartbeat.*"
+)
+
+var _ Service = (*register)(nil)
+
+type Service interface {
+	Applications() map[string]*Application
+}
+
+type register struct {
+	nc   *nats.Conn
+	apps map[string]*Application
+}
+
+func New(nc *nats.Conn) (Service, error) {
+	r := register{
+		nc:   nc,
+		apps: make(map[string]*Application),
+	}
+
+	_, err := r.nc.Subscribe(HEARTBEAT, func(msg *nats.Msg) {
+		sub := msg.Subject
+		appname := strings.Split(sub, ".")[1]
```

Comment:
Please add length check before accessing by index.

Reply:
ok, though Heartbeat is constant that has at least two elements